boards/ek-tm4c1294xl/src/io.rs

```diff
@@ -0,0 +1,120 @@
+use core::fmt::*;
+use kernel::hil::uart::{self, UART};
+use kernel::process;
+use tm4c129x;
+
+pub struct Writer {
+    initialized: bool,
+}
+
+pub static mut WRITER: Writer = Writer { initialized: false };
+
+impl Write for Writer {
+    fn write_str(&mut self, s: &str) -> ::core::fmt::Result {
+        let uart = unsafe { &mut tm4c129x::uart::UART0 };
+        if !self.initialized {
+            self.initialized = true;
+            uart.init(uart::UARTParams {
+                baud_rate: 115200,
+                stop_bits: uart::StopBits::One,
+                parity: uart::Parity::None,
+                hw_flow_control: false,
+            });
+            unsafe {
+                uart.specify_pins(&tm4c129x::gpio::PA[0], &tm4c129x::gpio::PA[1]);
+            }
+            uart.enable_tx();
+        }
+        // XXX: I'd like to get this working the "right" way, but I'm not sure how
+        for c in s.bytes() {
+            uart.send_byte(c);
+            while !uart.tx_ready() {}
+        }
+        Ok(())
+    }
+}
+
+#[cfg(not(test))]
+#[no_mangle]
+#[lang = "panic_fmt"]
+pub unsafe extern "C" fn panic_fmt(args: Arguments, file: &'static str, line: u32) -> ! {
```

Comment:
I believe this can be simplified now. See https://github.com/helena-project/tock/blob/master/boards/hail/src/io.rs#L40

Reply:
Ok, thanks for the hint.